src/components/QuranReader/index.tsx

```diff
@@ -49,47 +38,10 @@ const QuranReader = ({
   const isSidebarNavigationVisible = useSelector(selectIsSidebarNavigationVisible);
   const readingPreference = useSelector(selectReadingPreference) as ReadingPreference;
   const isReadingPreference = readingPreference === ReadingPreference.Reading;
-  const chaptersData = useContext(DataContext);
-  const dispatch = useDispatch();
-  const { cache } = useSWRConfig();
-
-  const addReadingSessionAndClearCache = useCallback(
-    (chapterNumber, verseNumber) => {
-      addReadingSession(chapterNumber, verseNumber).then(() => {
-        cache.delete(makeReadingSessionsUrl());
-      });
-    },
-    [cache],
-  );
-
-  const debouncedAddReadingSession = useMemo(
-    () => debounce(addReadingSessionAndClearCache, READING_SESSION_DEBOUNCE_WAIT_TIME),
-    [addReadingSessionAndClearCache],
-  );
 
-  const onElementVisible = useCallback(
-    (element: Element) => {
-      const lastReadVerse = getObservedVersePayload(element);
-      const [chapterNumber, verseNumber] = getVerseAndChapterNumbersFromKey(lastReadVerse.verseKey);
-      dispatch(
-        setLastReadVerse({
-          lastReadVerse,
-          chaptersData,
-        }),
-      );
-
-      if (isLoggedIn()) {
-        debouncedAddReadingSession(Number(chapterNumber), Number(verseNumber));
-      }
-    },
-    [chaptersData, debouncedAddReadingSession, dispatch],
-  );
```

Comment:
This logic should not be changed. reading sessions should be kept as they are as they are out of the scope of this feature.